packages/runner/src/run.ts

```diff
@@ -588,21 +599,35 @@ export async function runFiles(files: File[], runner: VitestRunner): Promise<voi
 }
 
 export async function startTests(specs: string[] | FileSpecification[], runner: VitestRunner): Promise<File[]> {
-  const paths = specs.map(f => typeof f === 'string' ? f : f.filepath)
-  await runner.onBeforeCollect?.(paths)
+  const cancel = runner.cancel?.bind(runner)
+  runner.cancel = (reason) => {
```

Comment:
Ideally, we _need_ to have an event listener for this, but only have a runner here. Adding another `onCancel` felt wrong (maybe it needs to be refactored)